src/Cake.Core/ITaskTeardownContext.cs

```diff
@@ -0,0 +1,25 @@
+using System;
+
+namespace Cake.Core
+{
+    public interface ITaskTeardownContext
+    {
+        ICakeTaskInfo Task { get; }
+        TimeSpan Duration { get; }
+        bool Skipped { get; }
+    }
+
+    public sealed class TaskTeardownContext : ITaskTeardownContext
+    {
+        public TaskTeardownContext(ICakeTaskInfo task, TimeSpan duration, bool skipped)
+        {
+            Task = task;
+            Duration = duration;
+            Skipped = skipped;
+        }
+
+        public ICakeTaskInfo Task { get; private set; }
```

Comment:
:+1: this is actually also my preferred personal coding style.
How do you guys feel about null checks / guard clauses for the args on these kinds of things (that's another thing I typically do)?